src/engine.rs

```diff
@@ -81,6 +82,7 @@ pub enum ResourceProxy {
 }
 
 pub enum ExternalResource<'a> {
+    #[allow(unused)]
```

Comment:
The unfortunate thing about allow unused is that there's no unused allow unused lint.
We just need to hope that we'll be vigilant enough to catch it when these variants do become used.